packages/atomic/src/components/result-template-components/atomic-size-condition/atomic-size-condition.tsx

```diff
@@ -0,0 +1,150 @@
+import {Element, Component, Prop, State} from '@stencil/core';
+
+type Criterion = (width: number, height: number) => boolean;
+
+/**
+ * Displays children only if their parent meets the defined size constraints.
+ */
+@Component({
+  tag: 'atomic-size-condition',
+  styleUrl: 'atomic-size-condition.pcss',
+  shadow: false,
+})
+export class AtomicSizeCondition {
+  private resizeObserver!: ResizeObserver;
+
+  @Element() private host!: HTMLElement;
+
+  @State() private shouldRender!: boolean;
+
+  /**
+   * The minimum width (exclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public minimumWidthExclusive?: string;
+  /**
+   * The minimum width (inclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public minimumWidthInclusive?: string;
+  /**
+   * The maximum width (exclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public maximumWidthExclusive?: string;
+  /**
+   * The maximum width (inclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public maximumWidthInclusive?: string;
+  /**
+   * The minimum height (exclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public minimumHeightExclusive?: string;
+  /**
+   * The minimum height (inclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public minimumHeightInclusive?: string;
+  /**
+   * The maximum height (exclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public maximumHeightExclusive?: string;
+  /**
+   * The maximum height (inclusively) required to display children.
+   *
+   * E.g.: `3rem`, `500px` or `30vw`.
+   */
+  @Prop() public maximumHeightInclusive?: string;
+
+  constructor() {
+    this.shouldRender = this.elementMeetsCriteria;
+  }
+
+  public componentDidLoad() {
+    this.resizeObserver = new ResizeObserver(
+      () => (this.shouldRender = this.elementMeetsCriteria)
+    );
+    this.resizeObserver.observe(this.parent);
```

Comment:
Should it observe the parent, or the global window/document ? 🤔 

Said differently, as a user, would would expect the `atomic-size-condition` component be dependant on the direct parent size, or the page itself ?

I think the parent is logical/fine by me, but wanted to know others opinions.

Reply:
The reason I made this component was to hide/show components when the visual section is smaller/larger, which is affected both by screen size and by the density & image size of the result.

We *could* add support for screen size, but since media queries can already do that it may be redundant.